src/components/todo-item/index.js

```diff
@@ -2,15 +2,48 @@ import React from 'react';
 
 export default ({
   remove = e => console.log( 'no remove method defined' ),
+  toggle = e => console.log( 'no toggle method defined' ),
```

Comment:
I just meant that updating the name and updating the completion status are really the same: updating the todo item. So you could just fire an update, passing in the relevant fields into the action. So you could combine two callbacks, two actions, and two cases in the reducer into one of each: update callback, `UPDATE_TODO` action, and one case in the reducer to handle it.